packages/cli/src/link/ios/getTargets.js

```diff
@@ -15,25 +15,31 @@ module.exports = function getTargets(project) {
     firstProject: { targets },
   } = project.getFirstProject();
   const nativeTargetSection = project.pbxNativeTargetSection();
-  return targets.map(target => {
-    const key = target.value;
-    const configurationListId = project.pbxNativeTargetSection()[key]
-      .buildConfigurationList;
-    const configurationList = project.pbxXCConfigurationList()[
-      configurationListId
-    ];
-    const buildConfigurationId = configurationList.buildConfigurations[0].value;
-    const buildConfiguration = project.pbxXCBuildConfigurationSection()[
-      buildConfigurationId
-    ];
-    return {
-      uuid: key,
-      target: nativeTargetSection[key],
-      name: nativeTargetSection[key].productReference_comment,
-      isTVOS:
-        (buildConfiguration.buildSettings.SDKROOT &&
-          buildConfiguration.buildSettings.SDKROOT.indexOf('appletv') !== -1) ||
-        false,
-    };
-  });
+  return targets
+    .map(target => {
+      const key = target.value;
+      // If not a valid target, skip.
+      if (!project.pbxNativeTargetSection()[key]) return undefined;
```

Comment:
Let's flip `filter` and `map` order, e.g:
```js
targets.filter(target => nativeTargetSection[target.value] !== undefined).map(...)
```

Reply:
It would be also great to elaborate a bit more on the comment itself. For example, I would say: "Filter out targets that are ....". I am trying to understand what exactly is an invalid target (if it's in targets section and not in the native targets section - what does it mean).

Reply:
I run into the problem whenever I have an aggregate target alongside the default ones created by the cli. From the looks of it, we only need to modify native targets when linking, so anything not in the Native Target section should be ignored.

Reply:
Oh, now I remember! Due to limitations of Xcode library, I had to enumerate native targets by getting an array of project targets first. But yeah, I wrongly assumed that all targets will be in that section, which is not the case.